snuba/datasets/__init__.py

```diff
@@ -0,0 +1,40 @@
+from snuba import settings
+
+
+class DataSet(object):
+    """
+    A DataSet defines the complete set of data sources, schemas, and
+    transformations that are required to:
+        - Consume, transform, and insert data payloads from Kafka into Clickhouse.
+        - Define how Snuba API queries are transformed into SQL.
+
+    This is the the initial boilerplate. schema and processor will come.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.SCHEMA = None
+        self.PROD = False
```

Comment:
Indeed this was a bit confusing for me as well in the original PR, see this discussion:
https://github.com/getsentry/snuba/pull/294#discussion_r281040797

To answer your questions:
a) _what is this property for_: treat a prod dataset differently than a test one, like allowing ensure_table_exists function to automatically create a missing table or to drop the existing one in tests and dev but not in prod.
b) _why is it on the dataset class instead of being an environment variable_: that's because in this architecture we have a class per dataset per environment, so whether a dataset is prod vs test is a static property of the dataset class. The environment variable drives which dataset class is used in a given environment (DATASETS property in the settings file). So this property is indirectly tied to the environment variable.
c) _why do we need one dataset class per environment instead of making the dataset class behave differently in different environments?_ Here I have to partially guess since this was from the original PR and I don't have full context on the reasoning there. I think in the original design Alex expected a lot of differences between the dataset classes. This turned out not to be the case (https://github.com/getsentry/snuba/pull/294/files#diff-4748d36d4c99330e47b82e88f38f5319R1009) so finally the table name is the only difference as of now. 

So finally I would be ok in revisiting the design and just removing the Dev and Test classes for the dataset relying instead on the application environment. Should I do it here or as a follow up ?

Reply:
Thinking over it, @tkaemming, unless the explanation above rang you a bell and it reminded you about some strong reason to keep separate classes, I will remove them in this PR so we change production config only once.